src/class_resolver/docs.py

```diff
@@ -0,0 +1,134 @@
+"""Utilities for automatic documentation."""
+
+from __future__ import annotations
+
+import inspect
+import textwrap
+from typing import Callable, TypeVar
+
+__all__ = [
+    "document_resolver",
+]
+
+F = TypeVar("F", bound=Callable)
+
+
+def document_resolver(
+    *params: str | tuple[str, str],
```

Comment:
I want to not use variadics here so we don't have to use keywords for the second argument

Reply:
What about changing the order (and still using variadics)? This would allow to only use a single decorator for multiple arguments for the same resolver (not sure how command that is though :sweat_smile: ).

Reply:
That’s a good point but still missing the case when there are multiple different resolvers. I think we might need a separate named tuple or something like that to make it possible to define everything in a single decorator, which could have the benefit of grouping all of the notes together

Reply:
Sorry I suggested we upstream this then I’m holding it up but it will be a pain to change the interface later